Source/JavaScriptCore/assembler/MacroAssemblerARM64.h

```diff
@@ -3292,8 +3294,12 @@ class MacroAssemblerARM64 : public AbstractMacroAssembler<Assembler> {
 
     void move(RegisterID src, RegisterID dest)
     {
-        if (src != dest)
-            m_assembler.mov<64>(dest, src);
+        if (src != dest) {
+            if (src == ARM64Registers::zr && dest != ARM64Registers::sp)
```

Comment:
Why we cannot move `0` to `sp`?

Reply:
sp is special register, and we cannot use as a target of movz on ARM64.